plugins/needs_restarting.py

```diff
@@ -100,6 +101,18 @@ def print_cmd(pid):
     print('%d : %s' % (pid, command))
 
 
+def get_service(pid):
+    cmdline = 'ps -o unit= %d' % pid
+    process = subprocess.Popen(['ps', '-o', 'unit=', str(pid)],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+    servicename = str(stdout, 'utf-8').rstrip()
+    if servicename.endswith(".service"):
+        return servicename
+    return
```

Comment:
I think some error handling should be in place here? At least check the exit status for the `ps` call and possibly also if the service doesn't match? Not sure what can possibly happen here and how it should be handled but I think it should be handled somehow :slightly_smiling_face: